api/api.go

```diff
@@ -318,18 +324,220 @@ func (f *PublicController) Register(c *gin.Context) {
 func (f *PublicController) GetAllAppointments(c *gin.Context) {
 
 	user_id, err := strconv.Atoi(c.Param("user_id"))
+
+		appointments, err := calender.GetAllAppointments(f.Database, user_id)
+	if err != nil {
+		log.Errorf("Unable to get appointments from user: %s\n", err.Error())
+		c.IndentedJSON(http.StatusBadRequest, err.Error())
+		return
+	}
+	c.Header("Access-Control-Allow-Origin", "*")
+	c.IndentedJSON(http.StatusOK, appointments)
+}
+*/
+func (f *PublicController) GetUserCount(c *gin.Context) {
+
+	//Fetch Data from Database with Backend function
+	usercount, err := institution.GetUserCount(f.Database)
+	if err != nil {
+		log.Errorf("Unable to get usercount: %s\n", err.Error())
+		c.IndentedJSON(http.StatusBadRequest, err.Error())
+		return
+	}
+	//Return Status and Data in JSON-Format
+	c.Header("Access-Control-Allow-Origin", "*")
+	c.IndentedJSON(http.StatusOK, usercount)
+
+}
+
+func (f *PublicController) GetAllFieldsOfStudy(c *gin.Context) {
+
+	//Fetch Data from Database with Backend function
+	fieldofstudies, err := institution.GetAllFieldsOfStudies(f.Database)
+	if err != nil {
+		log.Errorf("Unable to get all FieldOfStudies: %s\n", err.Error())
+		c.IndentedJSON(http.StatusBadRequest, err.Error())
+		return
+	}
+	//Return Status and Data in JSON-Format
+	c.Header("Access-Control-Allow-Origin", "*")
+	c.IndentedJSON(http.StatusOK, fieldofstudies)
+
+}
+
+func (f *PublicController) CreateFieldOfStudy(c *gin.Context) {
+
+	raw, err := c.GetRawData()
 	if err != nil {
+		log.Errorf("Unable to get raw data from request: %s\n", err.Error())
+		c.IndentedJSON(http.StatusBadRequest, err.Error())
+		return
+	}
+
+	var j map[string]interface{}
+	err = json.Unmarshal(raw, &j)
+	if err != nil {
+		log.Errorf("Unable to unmarshal the json body: %+v", raw)
+		c.Status(http.StatusInternalServerError)
+		return
+	}
+
+	name, ok := j["name"].(string)
+	if !ok {
+		log.Error("unable to convert name to string")
+		c.Status(http.StatusInternalServerError)
+		return
+	}
+
+	tmp, ok := j["semesters"].(float64)
+	if !ok {
+		log.Error("unable to convert semesters to float64")
 		c.Status(http.StatusInternalServerError)
```

Comment:
Das wäre dann eher ein `http.StatusBadRequest`, da der Wert vom User kommt und wenn der Wert nicht in einen `float64` umgewandelt werden kann es am User Input liegt.